imap/append.c

```diff
@@ -908,6 +935,30 @@ EXPORTED int append_fromstage_full(struct appendstate *as, struct body **body,
         if (r) goto out;
     }
 
+    if (!nolink) {
+        /* attempt to find an existing message with the same guid
+           and use it as the stagefile */
+        struct conversations_state *cstate, *mycstate = NULL;
+
+        cstate = conversations_get_mbox(mailbox->name);
+        if (!cstate) {
+            r = conversations_open_mbox(mailbox->name, 1/*shared*/, &mycstate);
```

Comment:
This will need to be 0/*shared*/ because we're planning to write to the mailbox.  I think we can just use:

```cstate = mailbox_get_cstate(mailbox);```

and avoid all the mycstate guff.  It will get the correct type anyway, given that the mailbox will be opened read-write.